Can't mock Send traits

mock-it cannot mock a Trait that is `Send`.  Example:
```rust
    fn send() {
        pub trait A {
            fn foo(&self);
        }
        
        struct AMock {
            foo: Mock<(), ()>
        }
        impl A for AMock {
            fn foo(&self) {
                self.foo.called(())
            }
        }
        impl AMock {
            fn new() -> AMock {
                AMock {
                    foo: Mock::new()
                }
            }
        }
        
        let mock = AMock::new();
        let _ = Box::new(mock) as Box<A + Send>;
    }
```
This results in the error
```
error[E0277]: `std::rc::Rc<std::cell::RefCell<std::vec::Vec<()>>>` cannot be sent between threads safely
   --> src/t_mock_it.rs:450:17
    |
450 |         let _ = Box::new(mock) as Box<A + Send>;
    |                 ^^^^^^^^^^^^^^ `std::rc::Rc<std::cell::RefCell<std::vec::Vec<()>>>` cannot be sent between threads safely                                   
    |
    = help: within `<t_mock_it::t::MockIt as TestSuite>::send::AMock`, the trait `std::marker::Send` is not implemented for `std::rc::Rc<std::cell::RefCell<std::vec::Vec<()>>>`
    = note: required because it appears within the type `mock_it::Mock<(), ()>`
    = note: required because it appears within the type `<t_mock_it::t::MockIt as TestSuite>::send::AMock`
    = note: required for the cast to the object type `dyn <t_mock_it::t::MockIt as TestSuite>::send::A + std::marker::Send`
```

--- a/src/mock.rs
+++ b/src/mock.rs
@@ -1,13 +1,13 @@
 use output;
 use output::Output;
 use rule::Rule;
-use std::cell::RefCell;
-use std::rc::Rc;
+use std::sync::Arc;
+use std::sync::Mutex;
 use validator::*;
 
 pub struct Mock<I, O> {
-    calls: Rc<RefCell<Vec<I>>>,
-    rules: Rc<RefCell<Vec<Rule<I, Output<O>>>>>,
+    calls: Arc<Mutex<Vec<I>>>,
+    rules: Arc<Mutex<Vec<Rule<I, Output<O>>>>>,
     default: O,
 }
 
@@ -24,34 +24,35 @@ impl<I, O: Clone> Clone for Mock<I, O> {
 impl<I: PartialEq, O: Clone> Mock<I, O> {
     pub fn new(default: O) -> Mock<I, O> {
         Mock {
-            calls: Rc::new(RefCell::new(Vec::new())),
-            rules: Rc::new(RefCell::new(Vec::new())),
+            calls: Arc::new(Mutex::new(Vec::new())),
+            rules: Arc::new(Mutex::new(Vec::new())),
             default: default,
         }
     }
 
     pub fn given(&self, input: I) -> Output<O> {
-        let return_value = Rc::new(RefCell::new(self.default.clone()));
+        let return_value = Arc::new(Mutex::new(self.default.clone()));
         self.rules
-            .borrow_mut()
+            .lock()
+            .unwrap()
             .push(Rule::new(input, Output::new(return_value.clone())));
         Output::new(return_value)
     }
 
     pub fn called(&self, input: I) -> O {
-        for value in &*self.rules.borrow() {
+        for value in &*self.rules.lock().unwrap() {
             if &value.input == &input {
-                self.calls.borrow_mut().push(input);
+                self.calls.lock().unwrap().push(input);
                 return output::value_of(value.output.clone());
             }
         }
 
-        self.calls.borrow_mut().push(input);
+        self.calls.lock().unwrap().push(input);
         return self.default.clone();
     }
 
     pub fn was_called_with(&self, input: I) -> Validator<I> {
-        for value in &*self.calls.borrow() {
+        for value in &*self.calls.lock().unwrap() {
             if value == &input {
                 return Validator::new(self.calls.clone(), true, input);
             }

--- a/src/output.rs
+++ b/src/output.rs
@@ -1,25 +1,26 @@
-use std::rc::Rc;
-use std::cell::RefCell;
+use std::sync::Arc;
+use std::sync::Mutex;
 
 #[derive(Clone)]
 pub struct Output<O> {
-    value: Rc<RefCell<O>>,
+    value: Arc<Mutex<O>>,
 }
 
 impl<O> Output<O> {
-    pub fn new(value: Rc<RefCell<O>>) -> Output<O> {
+    pub fn new(value: Arc<Mutex<O>>) -> Output<O> {
         Output { value: value }
     }
 }
 
 impl<O: Clone> Output<O> {
     pub fn will_return(&self, value: O) {
-        *self.value.borrow_mut() = value;
+        let mut tmp = self.value.lock().unwrap();
+        *tmp = value;
     }
 }
 
 pub fn value_of<O: Clone>(output: Output<O>) -> O {
-    output.value.borrow().clone()
+    output.value.lock().unwrap().clone()
 }
 
 #[cfg(test)]
@@ -30,7 +31,7 @@ mod test {
     fn given_default_output_when_will_return_expected_then_should_be_expected() {
         let default = 0;
         let expected = 5;
-        let output = Output::new(Rc::new(RefCell::new(default)));
+        let output = Output::new(Arc::new(Mutex::new(default)));
 
         output.will_return(expected);
         let actual = value_of(output);
@@ -40,7 +41,7 @@ mod test {
     #[test]
     fn given_default_output_when_dont_call_will_return_then_should_be_default() {
         let default = 0;
-        let output = Output::new(Rc::new(RefCell::new(default)));
+        let output = Output::new(Arc::new(Mutex::new(default)));
 
         let actual = value_of(output);
 

--- a/src/validator.rs
+++ b/src/validator.rs
@@ -1,18 +1,18 @@
-use std::cell::RefCell;
-use std::rc::Rc;
+use std::sync::Arc;
+use std::sync::Mutex;
 
 pub fn verify<I>(validator: Validator<I>) -> bool {
     validator.result
 }
 
 pub struct Validator<I> {
-    calls: Rc<RefCell<Vec<I>>>,
+    calls: Arc<Mutex<Vec<I>>>,
     result: bool,
     input: I,
 }
 
 impl<I: PartialEq> Validator<I> {
-    pub fn new(calls: Rc<RefCell<Vec<I>>>, result: bool, input: I) -> Validator<I> {
+    pub fn new(calls: Arc<Mutex<Vec<I>>>, result: bool, input: I) -> Validator<I> {
         Validator {
             calls: calls,
             result: result,
@@ -26,7 +26,7 @@ impl<I: PartialEq> Validator<I> {
         }
 
         let mut counter = 0;
-        for value in &*self.calls.borrow() {
+        for value in &*self.calls.lock().unwrap() {
             if value == &self.input {
                 counter += 1;
             }
@@ -45,7 +45,7 @@ mod test {
     #[test]
     #[should_panic]
     fn times_called_with_zero_should_panic() {
-        let validator = Validator::new(Rc::new(RefCell::new(Vec::new())), true, 5);
+        let validator = Validator::new(Arc::new(Mutex::new(Vec::new())), true, 5);
         validator.times(0);
     }
 
@@ -62,7 +62,7 @@ mod test {
 
         for (test_case, (calls, initial_result, input, times), expected) in table_test!(table) {
             let validator =
-                Validator::new(Rc::new(RefCell::new(calls.clone())), initial_result, input);
+                Validator::new(Arc::new(Mutex::new(calls.clone())), initial_result, input);
 
             let actual = verify(validator.times(times));
 

